sql/driver/conn.go

```diff
@@ -178,13 +178,18 @@ func (c *conn) Send(call sqlwire.Call) (*rows, error) {
 			} else if datum.IntVal != nil {
 				t[j] = *datum.IntVal
 			} else if datum.UintVal != nil {
-				t[j] = *datum.UintVal
+				// uint64 not supported by the driver.Value interface.
+				t[j] = int64(*datum.UintVal)
 			} else if datum.FloatVal != nil {
 				t[j] = *datum.FloatVal
 			} else if datum.BytesVal != nil {
 				t[j] = datum.BytesVal
 			} else if datum.StringVal != nil {
-				t[j] = *datum.StringVal
+				t[j] = []byte(*datum.StringVal)
+			}
+			if !driver.IsScanValue(t[j]) {
+				err := fmt.Errorf("Value of bad type %T returned by database", t[j])
```

Comment:
Error messages should not be capitalized: `s/Value/value/g`. I know this is called out in one of the go docs, but I can't seem to find it right now. 

Btw, the error message is a bit misleading. It isn't the database that is problematic, but the driver which did not convert the value appropriately. A panic might be a better option here as "this shouldn't ever happen".